### PR TITLE
 store/helper, inforschema: add mem table to show region info

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Add mem table to show information of `regions`.

### What is changed and how it works?

Using PD api. Since `peers` are useful. Split to `TIKV_REGION_STATUS` and `TIKV_REGION_PEER` two table.

```
MySQL [information_schema]> select * from tikv_region_peers limit 10;
+-----------+--------------------------------------------------------+--------------------------------------------------------+---------+----------+------------+-----------+--------+--------------+
| REGION_ID | START_KEY                                              | END_KEY                                                | PEER_ID | STORE_ID | IS_LEARNER | IS_LEADER | STATUS | DOWN_SECONDS |
+-----------+--------------------------------------------------------+--------------------------------------------------------+---------+----------+------------+-----------+--------+--------------+
|      3373 | 7480000000000000FF575F728000000009FFE70C020000000000FA | 7480000000000000FF575F72800000000AFF04DEE30000000000FA |    3374 |        1 |          1 |         0 | NORMAL |         NULL |
|      3373 | 7480000000000000FF575F728000000009FFE70C020000000000FA | 7480000000000000FF575F72800000000AFF04DEE30000000000FA |    3375 |        4 |          0 |         0 | NORMAL |         NULL |
|      3373 | 7480000000000000FF575F728000000009FFE70C020000000000FA | 7480000000000000FF575F72800000000AFF04DEE30000000000FA |    3376 |        5 |          0 |         0 | NORMAL |         NULL |
|      6597 | 7480000000000000FF595F72800000000DFF8A5F240000000000FA | 7480000000000000FF595F72800000000DFF9061190000000000FA |    6598 |        1 |          0 |         0 | NORMAL |         NULL |
|      6597 | 7480000000000000FF595F72800000000DFF8A5F240000000000FA | 7480000000000000FF595F72800000000DFF9061190000000000FA |    6599 |        4 |          0 |         0 | NORMAL |         NULL |
|      6597 | 7480000000000000FF595F72800000000DFF8A5F240000000000FA | 7480000000000000FF595F72800000000DFF9061190000000000FA |    6600 |        5 |          1 |         0 | NORMAL |         NULL |
|      7385 | 7480000000000000FF595F728000000010FFD3799C0000000000FA | 7480000000000000FF595F728000000010FFD97BA70000000000FA |    7386 |        1 |          0 |         0 | NORMAL |         NULL |
|      7385 | 7480000000000000FF595F728000000010FFD3799C0000000000FA | 7480000000000000FF595F728000000010FFD97BA70000000000FA |    7387 |        4 |          1 |         0 | NORMAL |         NULL |
|      7385 | 7480000000000000FF595F728000000010FFD3799C0000000000FA | 7480000000000000FF595F728000000010FFD97BA70000000000FA |    7388 |        5 |          0 |         0 | NORMAL |         NULL |
|      2993 | 7480000000000000FF575F728000000006FFC1CAC40000000000FA | 7480000000000000FF575F728000000006FFDF9DC00000000000FA |    2994 |        1 |          0 |         0 | NORMAL |         NULL |
+-----------+--------------------------------------------------------+--------------------------------------------------------+---------+----------+------------+-----------+--------+--------------+
10 rows in set (0.15 sec)

MySQL [information_schema]> select * from tikv_region_status limit 10;
+------+--------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+----------------+---------------+---------------+------------+------------------+------------------+
| ID   | START_KEY                                                                                  | END_KEY                                                                                    | EPOCH_CONF_VER | EPOCH_VERSION | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE | APPROXIMATE_KEYS |
+------+--------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+----------------+---------------+---------------+------------+------------------+------------------+
| 2065 | 7480000000000000FF555F728000000000FF05F37B0000000000FA                                     | 7480000000000000FF555F728000000000FF0BE6D00000000000FA                                     |              5 |           193 |             0 |          0 |               97 |           409600 |
| 2541 | 7480000000000000FF535F698000000000FF0000010380000000FF0020F58103800000FF00000088C6000000FC | 7480000000000000FF535F698000000000FF0000010380000000FF00249F0103800000FF00000061AE000000FC |              5 |           202 |             0 |          0 |               68 |           974832 |
| 4553 | 7480000000000000FF595F728000000005FF088B720000000000FA                                     | 7480000000000000FF595F728000000005FF0E93880000000000FA                                     |              5 |           407 |             0 |          0 |               95 |           368640 |
|  352 | 7480000000000000FF475F728000000000FF4051290000000000FA                                     | 7480000000000000FF475F728000000000FF465C620000000000FA                                     |              5 |            45 |             0 |          0 |               94 |           368640 |
|  672 | 7480000000000000FF475F698000000000FF0000010380000000FF0125076303800000FF0000000002000000FC | 7480000000000000FF475F698000000000FF0000010380000000FF0133B18003800000FF0000000002000000FC |              5 |            56 |             0 |          0 |               66 |           942080 |
|  692 | 7480000000000000FF475F728000000001FF442B400000000000FA                                     | 7480000000000000FF475F728000000001FF4A36520000000000FA                                     |              5 |            88 |             0 |          0 |               97 |           408736 |
| 2061 | 7480000000000000FF595F728000000000FF03CAC30000000000FA                                     | 7480000000000000FF595F728000000000FF09D8E20000000000FA                                     |              5 |           194 |             0 |          0 |               95 |           368640 |
| 7317 | 7480000000000000FF595F728000000010FF8B60F80000000000FA                                     | 7480000000000000FF595F728000000010FF9162E60000000000FA                                     |              5 |           897 |             0 |          0 |               97 |           409600 |
|  656 | 7480000000000000FF475F728000000001FF1FE8EC0000000000FA                                     | 7480000000000000FF475F728000000001FF25F40B0000000000FA                                     |              5 |            82 |             0 |          0 |               97 |           409600 |
| 2633 | 7480000000000000FF595F698000000000FF0000010380000000FF0089F40603800000FF0000000005000000FC | 7480000000000000FF595F698000000000FF0000010380000000FF00989A2303800000FF0000000003000000FC |              5 |           204 |            35 |          0 |               69 |           979802 |
+------+--------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+----------------+---------------+---------------+------------+------------------+------------------+
10 rows in set (0.11 sec)
```

**NOTE**:
PD API like `region topconfver/topread/topwrite/topsize/topversion` can be done by `ORDER BY .. LIMIT ..` CLAUSE.

### Check List 

Tests 

 - Unit test
 - Manual test (tested at a existing cluster got above results)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to be included in the release note

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -573,6 +576,118 @@ var tableAnalyzeStatusCols = []columnInfo{
 	{"STATE", mysql.TypeVarchar, 64, 0, nil, nil},
 }
 
+var tableTiKVRegionStatusCols = []columnInfo{
+	{"ID", mysql.TypeLonglong, 21, 0, nil, nil},
+	{"START_KEY", mysql.TypeVarchar, 64, 0, nil, nil},
```
Comment (lamxTyler) [inline infoschema/tables.go]: Is 64 long enough for key?

File: infoschema/tables.go [hunk 2]
```diff
@@ -573,6 +576,118 @@ var tableAnalyzeStatusCols = []columnInfo{
...
+
+var tableTiKVRegionPeersCols = []columnInfo{
+	{"REGION_ID", mysql.TypeLonglong, 21, 0, nil, nil},
+	{"START_KEY", mysql.TypeVarchar, 64, 0, nil, nil},
+	{"END_KEY", mysql.TypeVarchar, 64, 0, nil, nil},
+	{"PEER_ID", mysql.TypeLonglong, 21, 0, nil, nil},
+	{"STORE_ID", mysql.TypeLonglong, 21, 0, nil, nil},
+	{"IS_LEARNER", mysql.TypeTiny, 1, mysql.NotNullFlag, 0, nil},
+	{"IS_LEADER", mysql.TypeTiny, 1, mysql.NotNullFlag, 0, nil},
+	{"STATUS", mysql.TypeString, 1, 0, 0, nil},
```
Comment (lamxTyler) [inline infoschema/tables.go]: Is `1` enough?

File: store/helper/helper.go [hunk 3]
```diff
@@ -365,3 +366,78 @@ func (r *RegionFrameRange) GetIndexFrame(tableID, indexID int64, dbName, tableNa
 	}
 	return nil
 }
+
+// RegionPeer stores information of one peer.
+type RegionPeer struct {
+	ID        int64 `json:"id"`
+	StoreID   int64 `json:"store_id"`
```
[the remark this answers is not in the document]
Comment (winoros) [reply store/helper/helper.go]: This can be achieved by join with `TIKV_STORE_STATUS` table.

**Comment by morgo** [comment]
Is it possible to provide a helper function, or some way to decode a key such that:
```
7480000000000000FF575F728000000009FFE70C020000000000FA => {tableName, IndexName, [KeyValues]}
```
If the helper function returns JSON that would help for the case that there could be more than one value, and the values will differ in type.

**Comment by winoros** [comment]
@morgo Sure, i'll find a proper way to add it as a function.

**Pull request merged**